internal/provider/provider.go

```diff
@@ -119,25 +113,26 @@ func (p *PrefectProvider) Configure(ctx context.Context, req provider.ConfigureR
 		return
 	}
 
-	// Use provider values if supplied, otherwise fall back to environment variables
+	// Extract endpoint from configuration or environment variable.
+	// If the endpoint is not set, or the value is not a valid URL, emit an error.
 	var endpoint string
-	if !config.Endpoint.IsUnknown() && !config.Endpoint.IsNull() {
+	if !config.Endpoint.IsNull() {
 		endpoint = config.Endpoint.ValueString()
-	} else if u, ok := os.LookupEnv("PREFECT_API_URL"); ok {
-		endpoint = u
-	} else {
-		endpoint = "http://localhost:4200/api"
+	} else if apiURLEnvVar, ok := os.LookupEnv("PREFECT_API_URL"); ok {
+		endpoint = apiURLEnvVar
 	}
-
-	// Validate values (ensure that they are non-empty)
 	if endpoint == "" {
 		resp.Diagnostics.AddAttributeError(
 			path.Root("endpoint"),
 			"Missing Prefect API Endpoint",
 			"The Prefect API Endpoint is set to an empty value. "+
-				"Potential resolutions: set the endpoint attribute or PREFECT_API_URL environment variable to a non-empty value, or remove the value. "+fmt.Sprintf("endpoint %q unknown %t", endpoint, config.Endpoint.IsUnknown()),
+				"Potential resolutions: set the endpoint attribute or PREFECT_API_URL environment variable to a non-empty value, or remove the value.",
 		)
 	}
+	// Here, we'll ensure that the /api suffix is present on the endpoint
```

Comment:
this allows the end user to do something like:

```
provider "prefect" {
  endpoint = "https://api.prefect.cloud"
}

# instead of 
provider "prefect" {
  endpoint = "https://api.prefect.cloud/api"
}
```